account_move_import/wizard/import_move_line_wizard.py

```diff
@@ -0,0 +1,330 @@
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import base64
+import StringIO
+import csv
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+import logging
+_logger = logging.getLogger(__name__)
+
+header_fields = [
+    'account', 'debit', 'credit', 'name', 'partner', 'date_maturity',
+    'amount_currency', 'currency', 'tax_code', 'tax_amount',
+    'analytic_account']
+
+
+class aml_import(orm.TransientModel):
+    _name = 'aml.import'
+    _description = 'Import account move lines'
+    _columns = {
+        'aml_data': fields.binary('File', required=True),
+        'aml_fname': fields.char('Filename', size=128, required=True),
+        'csv_separator': fields.selection(
+            [(',', ','), (';', ';')], 'CSV Separator', required=True),
+        'decimal_separator': fields.selection(
+            [('.', '.'), (',', ',')], 'Decimal Separator', required=True),
+        'note': fields.text('Log'),
+    }
+    _defaults = {
+        'aml_fname': '',
+    }
+
+    def aml_import(self, cr, uid, ids, context=None):
+
+        account_obj = self.pool.get('account.account')
+        move_obj = self.pool.get('account.move')
+        partner_obj = self.pool.get('res.partner')
+        curr_obj = self.pool.get('res.currency')
+        tax_code_obj = self.pool.get('account.tax.code')
+        analytic_obj = self.pool.get('account.analytic.account')
+        mod_obj = self.pool.get('ir.model.data')
+        dp = self.pool.get('decimal.precision').precision_get(
+            cr, uid, 'Account')
+
+        company_id = context['company_id']
+        move_id = context['move_id']
+        account_ids = account_obj.search(
+            cr, uid, [('type', '!=', 'view'), ('company_id', '=', company_id)])
+        accounts = account_obj.read(cr, uid, account_ids, ['code', 'name'])
+
+        result_view = mod_obj.get_object_reference(
+            cr, uid, 'account_move_import', 'aml_import_result_view')
+        data = self.browse(cr, uid, ids)[0]
+        aml_file = data.aml_data
+        csv_separator = str(data.csv_separator)
+        decimal_separator = data.decimal_separator
+
+        err_log = ''
+        header_line = False
+
+        lines = base64.decodestring(aml_file)
+        reader = csv.reader(
```

Comment:
Generally these should be encapsulated in a with statement.

e.g.

```python
with StringIO.StringIO(lines) as lines:
```
As a side point, generally when using StringIO an import like

```python
try:
    import cStringIO as StringIO
except ImportError:
    import StringIO
```
and for code like this with potentially a lot of lines the extra performance might count.  Besides every OpenERP installation probably has cStringIO